### PR TITLE
gi: Add an override module and set basic magic methods

### PR DESCRIPTION
**Pull request by arteymix**

### CODE REVIEW

File: src/gi/overrides/Vast.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from ..module import get_introspection_module
+
+Vast = get_introspection_module('Vast')
+
+Vast.Array.__getitem__ = lambda self, i: self.get_value(i)
```
Comment (rainwoodman) [inline src/gi/overrides/Vast.py:5]: We need to refine this a bit in the future. If i is a slice or tuple of slices, return an array viewing the current array.
But what if it is a mixture of slice and indices? I don't think the current Array api support this; the dimensions indexed with a integer shall be eliminated.

Comment (rainwoodman) [reply src/gi/overrides/Vast.py:5]: I mean it is fine as of now because it is the vast.array api that needs to be fixed first.

Comment (arteymix) [reply src/gi/overrides/Vast.py:5]: Sure, we need to make a call based on the Python datatype, but for now we need to add the corresponding APIs in the array.

**Comment by rainwoodman** [comment]
I rebased this to master

**Comment by rainwoodman** [comment]
Merge? Master is broken anyways.

**Comment by arteymix** [comment]
There's still a lot to do. Maybe you would like to finish the magic methods first?

The override is not being imported so I'm sortof assuming it will work once installed.

**Comment by rainwoodman** [comment]
I want to first merge the infrastructure for override. Then we can play around what to override. We need that vala patch or a xslt transformation on the .gir file before this can actually work.

**Comment by rainwoodman** [comment]
But the if we don't merge the infrastructure first it needs to be rebased again and again and it is not trivial.

**Pull request merged**